### PR TITLE
fix(mcp): count preview threshold by runes

### LINKED ISSUE #751: bug(mcp): the [preview] marker counts bytes while the cut counts runes
### 📋 Pre-flight Checks

- [x] I have searched existing issues and this is not a duplicate
- [x] I understand this issue needs `status:approved` before a PR can be opened

### 📝 Bug Description

- `truncate()` cuts previews at 300 runes, but the `[preview]` marker and the "Results above are previews (300 chars)" footer trigger when content exceeds 300 BYTES.
- Non-ASCII content (accents, CJK, emoji) between 301 bytes and 300 runes is shown in full yet labeled `[preview]`.
- That label tells the agent to spend a `mem_get_observation` call on content it already has.

### 🔄 Steps to Reproduce

1. Save an observation whose content is 250 accented characters (over 300 bytes, under 300 runes).
2. Run `mem_search` and find it.
3. The full content is shown with `[preview]` appended, plus the preview footer.

### ✅ Expected Behavior

The marker appears only when the preview is actually truncated.

### ❌ Actual Behavior

The marker appears on content that is already complete.

### Operating System

macOS

### Engram Version

1.20.0 (`internal/mcp/mcp.go`, the `len(r.Content) > 300` check)

### Agent / Client

MCP (any agent)

### 💡 Additional Context

- One-line fix: `len([]rune(r.Content)) > 300`, matching the rune-based cut `truncate()` already does.
- Fix branch ready with a regression test. Will open the PR once this gets `status:approved`.

### PR DESCRIPTION
**Pull request by dnlrsls**
## 🔗 Linked Issue

Closes #751

---

## 🏷️ PR Type

- [x] `type:bug` — Bug fix
- [ ] `type:feature` — New feature
- [ ] `type:docs` — Documentation only
- [ ] `type:refactor` — Code refactoring (no behavior change)
- [ ] `type:chore` — Maintenance, dependencies, tooling
- [ ] `type:breaking-change` — Breaking change

---

## 📝 Summary

- Align MCP search preview labeling with the existing rune-based truncation boundary.
- Prevent complete multibyte content from being incorrectly marked as a preview.
- Cover both sides of the 300-rune boundary with handler-level regression tests.

## 📂 Changes

| File | Change |
|------|--------|
| `internal/mcp/mcp.go` | Count runes when deciding whether to append the preview marker and footer. |
| `internal/mcp/mcp_test.go` | Add runtime regressions for multibyte content below and above 300 runes. |

## 🧪 Test Plan

- [ ] Unit tests pass locally: `go test ./...`
- [ ] E2E tests pass locally: `go test -tags e2e ./internal/server/...`
- [ ] Manually tested the affected functionality
- [x] Focused regression passed: `go test ./internal/mcp -run '^TestHandleSearchPreviewUsesRuneCount$' -count=1`
- [x] Affected package passed: `go test ./internal/mcp -count=1`

The focused regression was first observed failing against the byte-count predicate, then passed after the rune-count correction. The affected-package run completed in 61.969 seconds.

---

## 🤖 Automated Checks

These run automatically and **all must pass** before merge:

| Check | What it verifies | Status |
|-------|-----------------|--------|
| **Check Issue Reference** | PR body contains `Closes #N` / `Fixes #N` / `Resolves #N` | ⏳ |
| **Check Issue Has status:approved** | Linked issue has `status:approved` label | ⏳ |
| **Check PR Has type:\* Label** | PR has exactly one `type:*` label | ⏳ |
| **Unit Tests** | `go test ./...` passes | ⏳ |
| **E2E Tests** | `go test -tags e2e ./internal/server/...` passes | ⏳ |
| **Plugin Tests** | `npm test` passes in `plugin/pi` | ⏳ |

---

## ✅ Contributor Checklist

- [x] I linked an approved issue above (`Closes #751`)
- [x] I added exactly **one** `type:*` label to this PR
- [ ] I ran unit tests locally: `go test ./...`
- [ ] I ran e2e tests locally: `go test -tags e2e ./internal/server/...`
- [ ] Docs updated (not required: this fixes labeling to match the existing documented 300-character preview behavior)
- [x] Commits follow [conventional commits](https://www.conventionalcommits.org/) format
- [x] No `Co-Authored-By` trailers in commits

---

## 💬 Notes for Reviewers

Receipt-driven review was requested, but the installed provider could not represent the target-bound STATUS continuation after START. The occurrence was recorded on the canonical provider tracker; this PR relies on the focused and affected-package verification reported above, with CI authoritative for repository-wide checks.

## Summary by CodeRabbit

* **Bug Fixes**
  * Search previews now correctly enforce the 300-character limit for content containing emoji, CJK characters, and other Unicode text.
  * Multibyte text is no longer truncated prematurely or counted incorrectly.
  * Content at or below the limit is displayed in full, while longer content is clearly marked as a preview.

### CODE REVIEW

File: internal/mcp/mcp_test.go [hunk 1]
```diff
@@ -1096,6 +1096,71 @@ func TestHelperArgsAndTruncate(t *testing.T) {
...
+			content:     strings.Repeat("é", 250),
+			wantContent: strings.Repeat("é", 250),
+			wantPreview: false,
+		},
+		{
+			name:        "multibyte content above rune limit is truncated",
+			content:     strings.Repeat("é", 301),
+			wantContent: strings.Repeat("é", 300) + "...",
+			wantPreview: true,
+		},
```
[the remark this answers is not in the document]
Comment (dnlrsls) [reply internal/mcp/mcp_test.go:1123]: Added the exact 300-rune multibyte boundary case in 06074a4. It asserts complete content with neither the preview marker nor footer, and the focused regression passes with `go test ./internal/mcp -run '^TestHandleSearchPreviewUsesRuneCount$' -count=1`.

**Comment by dnlrsls** [comment]
Closing this duplicate in favor of #856, which is the newer candidate for #751 and covers the same rune-count fix with stronger 200/300/301-rune boundary evidence. We will continue review there.
